--- dmsquash-generator.sh       2022-12-15 08:46:45.000000000 +0000
+++ dmsquash-generator.sh.new   2022-12-15 10:48:06.771676518 +0000
@@ -49,6 +49,12 @@ getargbool 0 rd.live.overlay.readonly -d
 getargbool 0 rd.live.overlay.overlayfs && overlayfs="yes"
 [ -e /xor_overlayfs ] && xor_overlayfs="yes"
 [ -e /xor_readonly ] && xor_readonly="--readonly"
+
+if [ "$overlayfs$xor_overlayfs" = "yes" ]; then
+  mkdir -p /run/overlayfs
+  mkdir -p /run/ovlwork
+fi
+
 ROOTFLAGS="$(getarg rootflags)"
 {
     echo "[Unit]"
